Classes/Domain/Model/Changes/AbstractStructuralChange.php

```diff
@@ -50,6 +50,33 @@ abstract class AbstractStructuralChange extends AbstractChange
      */
     protected $cachedSiblingNode = null;
 
+    /**
+     * Used when creating nodes within non-default tree preset
+     *
+     * @var string|null
+     */
+    protected $baseNodeType = null;
+
+    /**
+     * Set the baseNodeType
+     *
+     * @param string $baseNodeType
+     */
+    public function setBaseNodeType($baseNodeType)
```

Comment:
I don`t write that often PHP Code in the project but since all supported versions use modern PHP versions we should use typings.... don`t know if we have a rule for that but see @daniellienert is always adjusting that ;)